### DISCUSSION TITLE
Higher-level bitwise operations

### DISCUSSION THREAD

**Discussion by mdubinko** [toitlang/toit#10] [Ideas]
This seems like a good time and place to propose higher-level bitwise operations, of the sort often needed for device interfacing and the like. Not too long ago, I put together this (without Toit or any specific language in mind).

https://dubinko.info/blog/2021/10/low-level-bit-twiddling-in-high-level-languages/

Even if the proposed syntax doesn't work, this seems like an area ripe for innovation.

Welcome any thoughts...

**Comment by kasperl** [comment]
Thank you for these suggestions! How do you think it should look like in context of Toit?

Here is a simple example of low-level code with shifts and masks:
https://github.com/toitware/toit-lsm303dlh/blob/main/src/accelerometer.toit#L164

**Reply by mdubinko** [reply]
I'm all new to this language and might need some time to get familiar with it. Are there any plans to add destructuring assignment? That seems like a fairly natural way to accomplish something like this. There may be something interesting with toit-idiomatic blocks too. Hmm...

**Reply by floitsch** [reply]
There aren't any concrete plans, but such a feature is something we are considering.
It might take a bit of time before it reaches the top of our TODO lists, though.
